XCode/Sources/TlsSession.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  HttpRouter.swift
+//  Swifter
+//
+//  Copyright (c) 2014-2016 Damian Kołakowski. All rights reserved.
+//
+
+import Foundation
+
+#if !os(Linux)
+private func ensureNoErr(_ status: OSStatus) throws {
+    guard status == noErr else {
+        throw Errno.sslError(from: status)
+    }
+}
+
+public enum TLS {
+    /// Imports .p12 certificate file constructing structure to be used in TLS session.
+    ///
+    /// See [SecPKCS12Import](https://developer.apple.com/documentation/security/1396915-secpkcs12import).
+    /// Apple docs contain a misleading information that it does not import items to Keychain even though
+    /// it does.
+    ///
+    /// - Parameter _data: .p12 certificate file content
+    /// - Parameter password: password used when importing certificate
+    public static func loadP12Certificate(_ _data: Data, _ password: String) throws -> CFArray {
+        let data = _data as NSData
+        let options = [kSecImportExportPassphrase: password]
+        var items: CFArray!
+        try ensureNoErr(SecPKCS12Import(data, options as NSDictionary, &items))
+        let dictionary = (items! as [AnyObject])[0]
+        let secIdentity = dictionary[kSecImportItemIdentity] as! SecIdentity
```

Comment:
Same here we can please avoid the force-unwrapping even when Apple does in his examples?

Reply:
Well, this cannot quite be avoided, because when trying to make it optional compiler produces
`Conditional downcast to CoreFoundation type 'SecIdentity' will always succeed` error